snuba/cli/dead_letter_consumer.py

```diff
@@ -0,0 +1,124 @@
+import logging
+import signal
+from typing import Any, Optional, Sequence
+
+import click
+from arroyo import Topic, configure_metrics
+from arroyo.backends.kafka import KafkaConsumer
+from arroyo.processing import StreamProcessor
+
+from snuba import environment, settings
+from snuba.consumers.dead_letter_consumer import (
+    DeadLetterConsumerProcessingStrategyFactory,
+)
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_writable_storage
+from snuba.environment import setup_logging, setup_sentry
+from snuba.utils.metrics.wrapper import MetricsWrapper
+from snuba.utils.streams.configuration_builder import build_kafka_consumer_configuration
+from snuba.utils.streams.metrics_adapter import StreamMetricsAdapter
+from snuba.utils.streams.topics import Topic as StreamsTopic
+
+logger = logging.getLogger(__name__)
+
+
+@click.command(hidden=True)
+@click.option(
+    "--storage",
+    "storage_names",
+    type=click.Choice(["errors_v2", "transactions_v2"]),
+    multiple=True,
+    required=True,
+)
+@click.option(
+    "--consumer-group", default="snuba-dead-letter-consumers",
+)
+@click.option(
+    "--max-batch-size",
+    default=settings.DEFAULT_MAX_BATCH_SIZE,
+    type=int,
+    help="Max number of messages to batch in memory before writing to Kafka.",
+)
+@click.option(
+    "--max-batch-time-ms",
+    default=settings.DEFAULT_MAX_BATCH_TIME_MS,
+    type=int,
+    help="Max length of time to buffer messages in memory before writing to Kafka.",
+)
+@click.option(
+    "--auto-offset-reset",
+    default="error",
+    type=click.Choice(["error", "earliest", "latest"]),
+    help="Kafka consumer auto offset reset.",
+)
+@click.option(
+    "--queued-max-messages-kbytes",
+    default=settings.DEFAULT_QUEUED_MAX_MESSAGE_KBYTES,
+    type=int,
+    help="Maximum number of kilobytes per topic+partition in the local consumer queue.",
+)
+@click.option(
+    "--queued-min-messages",
+    default=settings.DEFAULT_QUEUED_MIN_MESSAGES,
+    type=int,
+    help="Minimum number of messages per topic+partition librdkafka tries to maintain in the local consumer queue.",
+)
+@click.option(
+    "--dead-letter-topic", help="Dead letter topic to consume failed insert messages."
+)
```

Comment:
For a separate PR, 
Since we are going to keep this around, topic names should not be passed as CLI arguments, they are registered in Snuba and provided by the storage via `KafkaStreamLoader` 
https://github.com/getsentry/snuba/blob/11ca1fd9cdb66657be3b82fda85d410a721b9ee9/snuba/datasets/table_storage.py#L104-L120

So using the CLI was ok during the emergency but we will have to express this in the same way as we do for other topics. Which means we need to think over what to do about multiple storages that use the same topic. Do we make one consumer take care of multiple storages or will we have one consumer per storage ?